src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -100,21 +100,27 @@ static int add_cryptsetup(
                         return log_oom();
         }
 
+        uki = efi_measured_uki(LOG_WARNING);
+
         if (measure) {
                 /* We only measure the root volume key into PCR 15 if we are booted with sd-stub (i.e. in a
                  * UKI), and sd-stub measured the UKI. We do this in order not to step into people's own PCR
                  * assignment, under the assumption that people who are fine to use sd-stub with its PCR
                  * assignments are also OK with our PCR 15 use here. */
-
-                r = efi_measured_uki(LOG_WARNING);
-                if (r == 0)
+                if (uki == 0)
                         log_debug("Will not measure volume key of volume '%s', not booted via systemd-stub with measurements enabled.", id);
-                else if (r > 0) {
+                else if (uki > 0) {
                         if (!strextend_with_separator(&options, ",", "tpm2-measure-pcr=yes"))
                                 return log_oom();
                 }
         }
 
+        if (uki > 0) {
+                /* Enable TPM2 based unlocking automatically, if we have a TPM. */
+                if (!strextend_with_separator(&options, ",", "tpm2-device=auto"))
+                        return log_oom();
+        }
```

Comment:
It is easier to follow if this is placed before the if branch for `measure`. Also, brackets are unnecessary.